pytorch_lightning/plugins/training_type/parallel.py

```diff
@@ -122,3 +127,21 @@ def block_backward_sync(self):
                 yield None
         else:
             yield None
+
+    def teardown(self) -> None:
+        """
+        This method is called to teardown the training process.
+        It is the right place to release memory and free other ressources.
+
+        By default, we teardown in the following way: if training is on gpu,
+        we move lightning module to CPU and clean up cuda memory.
+
+        At the end, we synchronize processes before returning control back to the caller.
```

Comment:
I'm not a fan of describing the code in text, as it is easy to forget to update and the code is quite clear already

Reply:
Fair point